### PR TITLE
feat(pt): add HybridMuonOptimizer

### PR DESCRIPTION
**Pull request by OutisLi**
## Summary by CodeRabbit

* **New Features**
  * Added HybridMuon optimizer option with configurable hyperparameters (momentum alias, lr_adjust, lr_adjust_coeff, min_2d_dim, muon_2d_only, weight decay) and integrated training support (initialization, LR tracking, scheduler compatibility, checkpoint resume).
* **Tests**
  * Added comprehensive tests covering orthogonalization, optimizer step behavior, weight-decay effects, fallback cases, LR-adjust modes, and state save/load.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: deepmd/pt/train/training.py [hunk 1]
```diff
@@ -726,6 +730,20 @@ def warm_up_linear(step: int, warmup_steps: int) -> float:
                 lr_adjust=float(self.opt_param.get("lr_adjust", 10.0)),
                 lr_adjust_coeff=float(self.opt_param.get("lr_adjust_coeff", 0.2)),
             )
+        elif self.opt_type == "Muon":
+            self.optimizer = MuonOptimizer(
```
Comment (njzjz) [inline deepmd/pt/train/training.py]: Why not use [`torch.optim.Muon`](https://docs.pytorch.org/docs/stable/generated/torch.optim.Muon.html), just like `Adam`?

Comment (wanghan-iapcm) [reply deepmd/pt/train/training.py]: If the current implementation deviates from the standard Muon optimizer architecture, I recommend renaming it to avoid confusion and clearly distinguish its unique functionality.

Comment (OutisLi) [reply deepmd/pt/train/training.py]: I coupled Muon with Adam, the native Muon do not have this. After my experiments, my realization is better in efficiency and performance.

File: deepmd/pt/train/training.py [hunk 2]
```diff
@@ -169,6 +171,10 @@ def get_opt_param(params: dict[str, Any]) -> tuple[str, dict[str, Any]]:
                 "momentum": params.get("momentum", 0.95),
                 "adam_beta1": params.get("adam_beta1", 0.9),
                 "adam_beta2": params.get("adam_beta2", 0.95),
+                "lr_adjust": params.get("lr_adjust", 10.0),
+                "lr_adjust_coeff": params.get("lr_adjust_coeff", 0.2),
+                "muon_2d_only": params.get("muon_2d_only", True),
+                "min_2d_dim": params.get("min_2d_dim", 1),
             }
```
Comment (njzjz) [inline deepmd/pt/train/training.py:178]: I don't find the meaning of this dict. It seems that `opt_param` is just a copy of `params`.

Comment (OutisLi) [reply deepmd/pt/train/training.py:178]: This requires removing the whole opt_param

**Review by wanghan-iapcm — APPROVED** [review]
